src/rust/engine/fs/src/lib.rs

```diff
@@ -745,7 +745,9 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
         expansion.todo,
       );
       // Finally, capture the resulting PathStats from the expansion.
-      expansion.outputs.into_iter().map(|(k, _)| k).collect()
+      let mut v: Vec<_> = expansion.outputs.into_iter().map(|(k, _)| k).collect();
+      v.sort_by(|a, b| a.path().cmp(b.path()));
```

Comment:
I figure that sorting glob expansion output is a thing we should probably be doing. Let me know if there's a strong reason not to.

I'd also somewhat like to include `foo` in the expansion of `foo/*` and `foo/**`, but I don't feel strongly enough about it to have implemented it without talking about it...

Reply:
We sort in `scandir`, which I _think_ is sufficient, recursively?

But as mentioned above: asserting that the inputs are sorted would be good.

Reply:
Introducing this sort introduced a bunch of test failures, so I guess it's important that we don't sort here. I didn't investigate why, but I can imagine reasons.

I'm not entirely convinced that sorting in the scandir is sufficient, so I moved the sorting to happen in fs_util rather than in VFS. Hopefully re-sorting a (hopefully small) already-sorted list isn't too expensive. If it is, we can fix it when it turns out to be.